stl/inc/numeric

```diff
@@ -469,6 +483,30 @@ _CONSTEXPR20 _OutIt adjacent_difference(const _InIt _First, const _InIt _Last, _
     const auto _ULast = _STD _Get_unwrapped(_Last);
     auto _UDest       = _STD _Get_unwrapped_n(_Dest, _STD _Idl_distance<_InIt>(_UFirst, _ULast));
     if (_UFirst != _ULast) {
+        if constexpr (_Iterators_are_contiguous<_InIt, _OutIt> && !_Iterator_is_volatile<_InIt>

```

Comment:
Perf bug: We should inspect the *unwrapped* iterators with `_Iterators_are_contiguous`:

https://github.com/microsoft/STL/blob/ca1553d38480b1224cb5b36ebd755af17aec275b/stl/inc/xutility#L4585-L4588

As written, C++14/17 code won't select the auto-vectorization codepath for anything but raw pointers; vector iterators won't be detected as contiguous. The benchmark didn't notice because it's always compiled as C++23.